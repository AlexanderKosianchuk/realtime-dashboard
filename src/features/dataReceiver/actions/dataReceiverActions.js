// @flow
import { createActions } from 'redux-actions';
import { DataReceiverActionTypes } from 'constants/index';

export const {
  dataReceiverConnect,
  dataReceiverDisconnect,
  dataReceiverSetStatus,
  dataReceiverSetData
} = createActions({
  [DataReceiverActionTypes.DATA_RECEIVER_CONNECT]: () => {},
  [DataReceiverActionTypes.DATA_RECEIVER_DISCONNECT]: () => {},
  [DataReceiverActionTypes.DATA_RECEIVER_SET_STATUS]: status => ({ status }),
  [DataReceiverActionTypes.DATA_RECEIVER_SET_DATA]: data => ({ data }),
});
