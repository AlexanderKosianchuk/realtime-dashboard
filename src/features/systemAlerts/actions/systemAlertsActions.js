import uid from 'nanoid';
import { createActions } from 'redux-actions';
import { AlertActionTypes } from 'constants/index';

export const { hideAlert, showAlert } = createActions({
  [AlertActionTypes.HIDE_ALERT]: (id: string) => ({ id }),
  [AlertActionTypes.SHOW_ALERT]: (message: string, options: Object) => {
    const timeout = options.variant === 'danger' ? 0 : 2;

    return {
      id: options.id || uid(),
      icon: options.icon,
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    };
  },
});
