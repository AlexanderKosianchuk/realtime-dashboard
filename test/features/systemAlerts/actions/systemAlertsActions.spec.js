import { hideAlert, showAlert } from 'features/systemAlerts/actions/systemAlertsActions';

describe('systemAlerts systemAlertsActions', () => {
  it('showAlert with variant `error` should return an action', () => {
    expect(showAlert('Alright!', { id: 'test', variant: 'danger' })).toMatchSnapshot();
  });

  it('showAlert with variant `success` should return an action', () => {
    expect(
      showAlert('Alright!', { id: 'test', variant: 'success', timeout: 10 }),
    ).toMatchSnapshot();
  });

  it('hideAlert should return an action', () => {
    expect(hideAlert('test')).toMatchSnapshot();
  });
});
