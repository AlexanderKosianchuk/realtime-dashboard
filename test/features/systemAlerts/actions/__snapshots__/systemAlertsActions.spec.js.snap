// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`systemAlerts systemAlertsActions hideAlert should return an action 1`] = `
Object {
  "payload": Object {
    "id": "test",
  },
  "type": "HIDE_ALERT",
}
`;

exports[`systemAlerts systemAlertsActions showAlert with variant \`error\` should return an action 1`] = `
Object {
  "payload": Object {
    "icon": undefined,
    "id": "test",
    "message": "Alright!",
    "position": "bottom-right",
    "timeout": 0,
    "variant": "danger",
  },
  "type": "SHOW_ALERT",
}
`;

exports[`systemAlerts systemAlertsActions showAlert with variant \`success\` should return an action 1`] = `
Object {
  "payload": Object {
    "icon": undefined,
    "id": "test",
    "message": "Alright!",
    "position": "bottom-right",
    "timeout": 10,
    "variant": "success",
  },
  "type": "SHOW_ALERT",
}
`;
